### DISCUSSION TITLE
Intercepting request via provider

### DISCUSSION THREAD

**Discussion by ninjz** [ethereum/web3.js#5827] [Q&A]
I'm looking for a way to intercept requests made through this library so I can capture the request being made to forward to Datadog to debug some slow network calls.

Is there any way to do this? I suspect creating a custom provider is somewhere in the right direction as I saw in this discussion: https://github.com/web3/web3.js/discussions/5025 but I'm kind of stuck from there as I don't see any way to do that within the `Web3.providers.HttpProvider` class.

**Comment by TatyOko28** [comment]
Yes, you can intercept and log requests made through Web3.js by creating a custom provider wrapper around HttpProvider.
 Solution: Extend Web3.providers.HttpProvider to log outgoing requests and forward them to Datadog for monitoring.
